src/cswinrt/strings/additions/Windows.Storage.Streams/WindowsRuntimeMarshal.cs

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace System.Runtime.InteropServices.WindowsRuntime
+{
+    using System.Diagnostics;
+    using System.Diagnostics.CodeAnalysis;
+    using System.IO;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using global::Windows.Foundation;
+    using global::Windows.Storage.Streams;
+    using WinRT;
+
+    /// <summary>
+    /// An unsafe class that provides a set of methods to access the underlying data representations of WinRT types.
+    /// </summary>
+#if EMBED || !NET // for netstandard this type conflicts with the type in the BCL so make it internal
+    internal
+#else
+    public
+#endif
+    static partial class WindowsRuntimeMarshal
+    {
+#if NET
+        private static global::System.Guid IID { get; } = new Guid(new global::System.ReadOnlySpan<byte>(new byte[] { 0xEF, 0x0F, 0x5A, 0x90, 0x53, 0xBC, 0xDF, 0x11, 0x8C, 0x49, 0x00, 0x1E, 0x4F, 0xC6, 0x86, 0xDA }));
+#else
+        private static global::System.Guid IID { get; } = new Guid(new byte[] { 0xEF, 0x0F, 0x5A, 0x90, 0x53, 0xBC, 0xDF, 0x11, 0x8C, 0x49, 0x00, 0x1E, 0x4F, 0xC6, 0x86, 0xDA });
+#endif
+        /// <summary>
+        /// Returns a pointer to the underlying data representation of the <see cref="IBuffer"/>.
+        /// Callers are responsible for ensuring that the buffer is kept alive while the pointer is in use.
+        /// </summary>
+        /// <param name="buffer">The buffer to get the data pointer for.</param>
+        /// <param name="dataPtr">The pointer to the underlying data representation of the buffer.</param>
+        public static unsafe bool TryGetDataUnsafe(IBuffer buffer, out IntPtr dataPtr)

```

Comment:
While we're at it, should we enable nullability annotations in this file? On `NET`, we could also make that `buffer` parameter be `[NotNullWhen(true)] IBuffer? buffer`, which could be kinda nice.